subsys/lorawan/nvm/lorawan_nvm_settings.c

```diff
@@ -0,0 +1,202 @@
+/*
+ * Copyright (c) 2022 Intellinium <giuliano.franchetto@intellinium.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <LoRaMac.h>
+#include <kernel.h>
+#include <settings/settings.h>
+#include "lorawan_nvm.h"
+#include <logging/log.h>
+
+LOG_MODULE_REGISTER(lorawan_nvm, CONFIG_LORAWAN_LOG_LEVEL);
+
+static uint16_t nvm_notify_flag;
+
+#define LORAWAN_SETTINGS_BASE                "lorawan/nvm"
+
+#define flag_set(f)        ((nvm_notify_flag & (f)) == (f))
+
+#define SAVE_SETTINGS(nvm, flag, name)					  \
+if (flag_set(flag)) {								  \
+	LOG_DBG("Saving configuration " LORAWAN_SETTINGS_BASE "/" # name);	  \
+	int err = settings_save_one(LORAWAN_SETTINGS_BASE "/" # name,	  \
+				&(nvm)->name,					  \
+				sizeof((nvm)->name));				  \
+	if (err) {								  \
+		LOG_ERR("Could not save settings" # name", error %d",	  \
+			err);							  \
+	}									  \
+}
+
+static void lorawan_nvm_save_settings(void)
+{
+	MibRequestConfirm_t mibReq;
+
+	LOG_DBG("Saving LoRaWAN settings");
+
+	/* Retrieve the actual context */
+	mibReq.Type = MIB_NVM_CTXS;
+	if (LoRaMacMibGetRequestConfirm(&mibReq) != LORAMAC_STATUS_OK) {
+		LOG_ERR("Could not get NVM context");
+		return;
+	}
+
+	LoRaMacNvmData_t *nvm = mibReq.Param.Contexts;
+
+	LOG_DBG("Crypto version: %"PRIu32", DevNonce: %d, JoinNonce: %"PRIu32,
+		mibReq.Param.Contexts->Crypto.LrWanVersion.Value,
+		mibReq.Param.Contexts->Crypto.DevNonce,
+		mibReq.Param.Contexts->Crypto.JoinNonce);
+
+	SAVE_SETTINGS(nvm, LORAMAC_NVM_NOTIFY_FLAG_CRYPTO, Crypto)
+	SAVE_SETTINGS(nvm, LORAMAC_NVM_NOTIFY_FLAG_MAC_GROUP1, MacGroup1)
+	SAVE_SETTINGS(nvm, LORAMAC_NVM_NOTIFY_FLAG_MAC_GROUP2, MacGroup2)
+	SAVE_SETTINGS(nvm, LORAMAC_NVM_NOTIFY_FLAG_SECURE_ELEMENT,
+		      SecureElement)
+	SAVE_SETTINGS(nvm, LORAMAC_NVM_NOTIFY_FLAG_REGION_GROUP1, RegionGroup1)
+	SAVE_SETTINGS(nvm, LORAMAC_NVM_NOTIFY_FLAG_REGION_GROUP2, RegionGroup2)
+	SAVE_SETTINGS(nvm, LORAMAC_NVM_NOTIFY_FLAG_CLASS_B, ClassB)
+
+	settings_save();
+	/* Clean notification flag */
+	nvm_notify_flag = LORAMAC_NVM_NOTIFY_FLAG_NONE;
+}
+
+void lorawan_nvm_data_mgmt_event(uint16_t notifyFlags)
```

Comment:
Also here: Please use correct coding style.
```suggestion
void lorawan_nvm_data_mgmt_event(uint16_t notify_flags)
```